walletnode/services/artworkregister/task_test.go

```diff
@@ -0,0 +1,519 @@
+				ListenOnSessID(testCase.args.primarySessID).
+				ListenOnAcceptedNodes(testCase.args.pastelIDS, testCase.args.acceptNodeErr)
+
+			nodes := node.List{}
+			for _, n := range testCase.args.nodes {
+				nodes.Add(node.NewNode(nodeClient.ClientMock, n.address, n.pastelID))
+			}
+
+			task := &Task{}
+			got, err := task.meshNodes(testCase.args.ctx, nodes, testCase.args.primaryIndex)
+
+			testCase.assertion(t, err)
+			assert.Equal(t, testCase.want, pullPastelAddressIDNodes(got))
+
+			nodeClient.RegArtWorkMock.AssertCalled(t, "AcceptedNodes", mock.Anything)
+			nodeClient.RegArtWorkMock.AssertNumberOfCalls(t, "AcceptedNodes", 1)
+			nodeClient.RegArtWorkMock.AssertCalled(t, "SessID")
+			nodeClient.RegArtWorkMock.AssertNumberOfCalls(t, "SessID", testCase.numSessIDCall)
+			nodeClient.RegArtWorkMock.AssertCalled(t, "Session", mock.Anything, false)
+			nodeClient.RegArtWorkMock.AssertCalled(t, "ConnectTo", mock.Anything, testCase.args.primaryPastelID, testCase.args.primarySessID)
+
+			nodeClient.ClientMock.AssertExpectations(t)
+			nodeClient.ConnectionMock.AssertExpectations(t)
+
+		})
+
+	}
+}
+
+func TestTaskIsSuitableStorageFee(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		Ticket *Ticket
+	}
+
+	type args struct {
+		ctx        context.Context
+		networkFee *pastel.StorageFee
+		returnErr  error
+	}
+
+	testCases := []struct {
+		fields    fields
+		args      args
+		want      bool
+		assertion assert.ErrorAssertionFunc
+	}{
+		{
+			fields: fields{
+				Ticket: &Ticket{MaximumFee: 0.5},
+			},
+			args: args{
+				ctx:        context.Background(),
+				networkFee: &pastel.StorageFee{NetworkFee: 0.49},
+			},
+			want:      true,
+			assertion: assert.NoError,
+		},
+		{
+			fields: fields{
+				Ticket: &Ticket{MaximumFee: 0.5},
+			},
+			args: args{
+				ctx:        context.Background(),
+				networkFee: &pastel.StorageFee{NetworkFee: 0.51},
+			},
+			want:      false,
+			assertion: assert.NoError,
+		},
+		{
+			args: args{
+				ctx:       context.Background(),
+				returnErr: fmt.Errorf("connection timeout"),
+			},
+			want:      false,
+			assertion: assert.Error,
+		},
+	}
+
+	for i, testCase := range testCases {
+		testCase := testCase
+
+		t.Run(fmt.Sprintf("testCase-%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			//create new mock service
+			pastelClient := pastelMock.NewMockClient()
+			pastelClient.ListenOnStorageFee(testCase.args.networkFee, testCase.args.returnErr)
+			service := &Service{
+				pastelClient: pastelClient.ClientMock,
+			}
+
+			task := &Task{
+				Service: service,
+				Ticket:  testCase.fields.Ticket,
+			}
+
+			got, err := task.isSuitableStorageFee(testCase.args.ctx)
+			testCase.assertion(t, err)
+			assert.Equal(t, testCase.want, got)
+
+			//pastelClient mock assertion
+			pastelClient.ClientMock.AssertExpectations(t)
+			pastelClient.ClientMock.AssertCalled(t, "StorageFee", testCase.args.ctx)
+			pastelClient.ClientMock.AssertNumberOfCalls(t, "StorageFee", 1)
```

Comment:
```suggestion
			pastelClient.AssertListenOnStorageFee(t, 1, testCase.args.ctx)
```

```go
func (c *Client)  AssertListenOnStorageFee(t TestingT, expectedCalls int , arguments ...interface{}) *Client {`
```

Can we keep it inside `test/mock_client.go` ? 
1. Not to repeat this in every unit test.
2. If we suddenly change the name of the method, e.g. `"StorageNetworkFee"` we need to change only in one place test/mock_client.go`
3. when it's all in one place, you can use a variable like
```go

type Client struct {
	*mocks.Client // not to repeat 
	nameStorageFee string
}

c.On(c.nameStorageFee, mock.Anything).Return(fee, returnErr)
...
c.AssertCalled(t, c.nameStorageFee, ctx)
...
```

One more, please read [this](https://gruntwork.io/guides/style%20guides/golang-style-guide/#naming-things) 
Although it is customary to give function names one or two letters, this will reduce readability.It would be great if you act like a rebel against these rules :) . e.g. Use `func (client *Client)` instead of `func (c *Client)`;

```go
// Client implementing pastel.Client for testing purpose
type Client struct {
	*mocks.Client
	nameMasterNodesTop string
}

// ListenOnMasterNodesTop listening MasterNodesTop and returns Mn's and error from args
func (client *Client) ListenOnMasterNodesTop(nodes pastel.MasterNodes, err error) *Client {
	client.On(client.nameMasterNodesTop, mock.Anything).Return(nodes, err)
	return client
}
```

Reply:
Got it